crates/autopilot/src/run_loop.rs

```diff
@@ -79,6 +78,10 @@ impl RunLoop {
     }
 
     async fn next_auction(&self) -> Option<domain::AuctionWithId> {
+        if let RunLoopMode::SyncToBlockchain = self.synchronization {
+            let _ = ethrpc::block_stream::next_block(self.eth.current_block()).await;
+        }
```

Comment:
There is also `tokio::time::sleep(Duration::from_secs(1)).await;` inside the `run_forever` function. Do we need to include this timeout only for `RunLoopMode::Unsynchronized`?

Reply:
Also, I might missed something.
Say, we started to process an auction at block 5 and finished on block 10, on the next iteration we will still be waiting for the next block 11. Shouldn't we start right away when we spent more than one block time for the previous auction?

Reply:
> There is also tokio::time::sleep(Duration::from_secs(1)).await;

Good catch!
Moved the logic into `run_forever` to either wait 1s or wait for the new block.
I moved the sleep before the run loop which will only have an effect on the very first auction but since building the first auction already takes more than 1s we should be good.

> Shouldn't we start right away when we spent more than one block time for the previous auction?

I think our efforts should be used to ensure that the runloop finishes shortly before the next block instead of expecting desyncs of the run loop.